components/action-mgt/org.wso2.carbon.identity.action.mgt/src/main/java/org/wso2/carbon/identity/action/mgt/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -0,0 +1,389 @@
+                }
+            }
+            return actions;
+        } catch (SQLException e) {
+            throw ActionManagementUtil.handleServerException(
+                    ActionMgtConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_RETRIEVING_ACTIONS_BY_ACTION_TYPE, e);
+        }
+    }
+
+    @Override
+    public Action updateAction(String actionType, String actionId, Action action,
+                               Integer tenantId) throws ActionMgtException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt = dbConnection.prepareStatement(
+                    ActionMgtSQLConstants.UPDATE_ACTION_BASIC_INFO)) {
+                prepStmt.setString(1, action.getName());
+                prepStmt.setString(2, action.getDescription());
+                prepStmt.setString(3, actionId);
+                prepStmt.setString(4, actionType);
+                prepStmt.setInt(5, tenantId);
+                prepStmt.executeUpdate();
+
+                // Update Endpoint Properties.
+                updateActionEndpointProperties(dbConnection, actionId, getEndpointProperties(action), tenantId);
+                IdentityDatabaseUtil.commitTransaction(dbConnection);
+
+                return getActionByActionId(actionId, tenantId);
+            } catch (SQLException | ActionMgtException e) {
+                IdentityDatabaseUtil.rollbackTransaction(dbConnection);
+                throw e;
+            }
+        } catch (SQLException e) {
+            throw ActionManagementUtil.handleServerException(
+                    ActionMgtConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_UPDATING_ACTION, e);
+        }
+    }
+
+    @Override
+    public void deleteAction(String actionType, String actionId, Integer tenantId) throws ActionMgtException {
+
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt = dbConnection.prepareStatement(ActionMgtSQLConstants.DELETE_ACTION)) {
+                prepStmt.setString(1, actionId);
+                prepStmt.setString(2, actionType);
+                prepStmt.setInt(3, tenantId);
+                prepStmt.executeUpdate();
+                IdentityDatabaseUtil.commitTransaction(dbConnection);
+            } catch (SQLException e) {
+                IdentityDatabaseUtil.rollbackTransaction(dbConnection);
+                throw e;
+            }
+        } catch (SQLException e) {
+            throw ActionManagementUtil.handleServerException(
+                    ActionMgtConstants.ErrorMessages.ERROR_CODE_ERROR_WHILE_DELETING_ACTION, e);
+        }
+    }
+
+    @Override
+    public Action activateAction(String actionType, String actionId, Integer tenantId) throws ActionMgtException {
+
+        changeActionStatus(actionType, actionId, ACTIVE_STATUS, tenantId);
+        return getActionByActionId(actionId, tenantId);
```

Comment:
This is another query. Do we really want to send the action on an activation and a deactivation
From API we just just refer the resource. I don't see the need to return the complete action actually

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5771/commits/94e90e62016b434a0f37aa2b1f7ff0edc34bb64f